systemFlake.nix

```diff
@@ -48,34 +53,42 @@ let
     "checksBuilder"
   ];
 
-  nixosConfigurationBuilder = hostname: profile: (
-    # It would be nice to get `nixosSystem` reference from `selectedNixpkgs` but it is not possible at this moment
-    inputs."${channelNameFromProfile profile}".lib.nixosSystem (genericConfigurationBuilder hostname profile)
-  );
-
   getNixpkgs = profile: self.pkgs."${systemFromProfile profile}"."${channelNameFromProfile profile}";
 
-  genericConfigurationBuilder = hostname: profile: (
-    let selectedNixpkgs = getNixpkgs profile; in
+  configurationBuilder = hostname: profile: (
+    let
+      selectedNixpkgs = getNixpkgs profile;
+      defaultBuilder =
+        if defaultConfigBuilder == null
+        then inputs."${channelNameFromProfile profile}".lib.nixosSystem
+        else defaultConfigBuilder;
+      builder = profile.builder or defaultBuilder;
+    in
     {
-      inherit (selectedNixpkgs) system;
-      modules = [
-        ({ pkgs, lib, ... }: {
-          networking.hostName = hostname;
-
-          nixpkgs = {
-            inherit (selectedNixpkgs) overlays config system;
-          };
-
-          system.configurationRevision = lib.mkIf (self ? rev) self.rev;
-          nix.package = lib.mkDefault pkgs.nixUnstable;
-        })
-      ]
-      ++ sharedModules
-      ++ (profile.modules or [ ]);
-      extraArgs = { inherit inputs; } // sharedExtraArgs // profile.extraArgs or { };
+      name = profile.output or defaultConfigOutput;
+      value.${hostname} = builder {
+        inherit (selectedNixpkgs) system;
+        modules = [
+          ({ pkgs, lib, ... }: {
+            networking.hostName = hostname;
+
+            nixpkgs = {
+              inherit (selectedNixpkgs) overlays config system;
+            };
+
+            system.configurationRevision = lib.mkIf (self ? rev) self.rev;
+            nix.package = lib.mkDefault pkgs.nixUnstable;
+          })
+        ]
+        ++ sharedModules
+        ++ (profile.modules or [ ]);
+        extraArgs = { inherit inputs; } // sharedExtraArgs // profile.extraArgs or { };
+      };
     }
   );
+
+  mapAttrs' = f: set:
+    builtins.listToAttrs (map (attr: f attr set.${attr}) (builtins.attrNames set));
```

Comment:
This is a copy from upstream, I don't know if you want to put this somewhere else, but I'm fairly certain its necessary to get the `output` option for profiles to work.

Reply:
Maybe we should create a separate file with all the copy-pastes from nixpkgs? It could be just straight up added to flake lib argumentnt

Reply:
@gytis-ivaskevicius what about putting it directly in `flake.nix`? since in staging systemflake now has access to flake-utils-plus lib.

Reply:
I was thinking about just dropping an import somewhere in here https://github.com/gytis-ivaskevicius/flake-utils-plus/blob/master/flake.nix#L26-L27 but adding it flake.nix directly isn't a bad idea, tho probably it does not scale well in the long run

Reply:
Wait isn't the first link also `flake.nix`, and thats what I was talking about.

Reply:
All i mean is seperation of sorts like 
```nix
      flake-utils.lib
      // (import ./copy-pasta.nix {} )
      // rec {
      ```

Reply:
ahh got it, I'll probably just add to flake.nix for now to avoid too many unrelated changes in this PR. But refactoring should definiitely happen